src/mdns/mdns.cpp

```diff
@@ -236,6 +238,12 @@ Publisher::TxtList Publisher::SortTxtList(TxtList aTxtList)
     return aTxtList;
 }
 
+Publisher::AddressList Publisher::SortAddressList(AddressList aAddressList)
+{
+    std::sort(aAddressList.begin(), aAddressList.end());
+    return aAddressList;
+}
```

Comment:
I guess you meant `void Publisher::SortAddressList(AddressList &aAddressList)`.

There are some reasons I made the method signature like this:
- It's more convenient to let it return the sorted list so that we can call `SortAddressList` in the member intializer list.
- This method signature follows the same style as `SortTxtList` and `SortSubTypeList`.
- There's no performance degradation to return the sorted list since we're using move constructors.

Reply:
The method prototype seems to imply that the original list is not modified, which is not the case?